fastapi/exceptions.py

```diff
@@ -15,11 +13,15 @@ def __init__(
         self.headers = headers
 
 
+RequestErrorModel = create_model("Request")
```

Comment:
pydantic now actually checks for the provided type to be a `BaseModel` so that it can handle pydantic dataclasses. Since we were only using the `Request` and `WebSocket` types to get the right name, I figured the `create_model` function provided the easiest way to get compatible models with the same name.

I would be open to changing the name, but since this results in the same error messages fastapi was spitting out before, I think this makes sense for now, and we can change the naming in a separate PR if desired.

Reply:
Cool! I think it's fine :+1: :rocket: